src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.health = health || 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
+  Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = hidden || false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+
+  bite(beast) {
+    if (beast.hidden === false && !(beast instanceof Carnivore)) {
+      beast.health = beast.health - 50;
+    }
+
+    if (beast.health === 0) {
+      Animal.alive = Animal.alive.filter(v => v !== beast);
```

Comment:
here you should delete beasts whose health is equal 0
that means you should change your condition for the filter method